vpr/src/route/route_timing.cpp

```diff
@@ -1230,14 +1237,14 @@ static bool timing_driven_route_sink(ClusterNetId net_id,
     t_bb bounding_box = route_ctx.route_bb[net_id];
 
     bool net_is_global = cluster_ctx.clb_nlist.net_is_global(net_id);
-    bool high_fanout = is_high_fanout(cluster_ctx.clb_nlist.net_sinks(net_id).size(), high_fanout_threshold);
+    bool high_fanout = is_high_fanout(cluster_ctx.clb_nlist.net_sinks(net_id).size(), router_opts.high_fanout_threshold);
     constexpr float HIGH_FANOUT_CRITICALITY_THRESHOLD = 0.9;
     bool sink_critical = (cost_params.criticality > HIGH_FANOUT_CRITICALITY_THRESHOLD);
 
     //We normally route high fanout nets by only adding spatially close-by routing to the heap (reduces run-time).
     //However, if the current sink is 'critical' from a timing perspective, we put the entire route tree back onto
     //the heap to ensure it has more flexibility to find the best path.
-    if (high_fanout && !sink_critical && !net_is_global) {
+    if (high_fanout && !sink_critical && !net_is_global && !(routing_predictor.get_slope() > router_opts.high_fanout_max_slope)) {
```

Comment:
As this logic gets more complicated (and it depends on more things like the RoutingPredictor), I wonder if we should hoist it out of timing_driven_route_sink and simply have a parameter which controls whether we do HF routing or not. I'm not sure which is better, so any thoughts are welcome.

Reply:
I prefer moving in the other direction, and letting optimized methods decide if they apply, and then several methods can be tried in order of priority.